src/validation/schema.js

```diff
@@ -339,9 +380,63 @@ const validateTypeDefinition = (defs, def) =>
 const validateTypeDefinitions = defs =>
   defs.reduce((errors, def) => errors.concat(validateTypeDefinition(defs, def)), List())
 
-const validateSchema = filename => {
-  let doc = loadSchema(filename)
+const validateNoOtherTypesExceptMutation = defs =>
+  defs.reduce((errors, def) =>
+    def.kind === "ObjectTypeDefinition" &&
+    def.name.value !== "Mutation"
+      ? errors.concat(immutable.fromJS([
+          {
+            loc: def.loc,
+            entity: def.name.value,
+            message:
+              `ObjectTypeDefinition '${def.name.value}' found, Mutation ` +
+              "must be the only type. Try using 'input' for mutation arguments, " +
+              "'interface' for mutation return values, or existing @entity types from the subgraph schema.",
```

Comment:
Let's unravel this a little bit. How about this error mesage?

```js
`\
Encountered object type '${def.name.value}. 'Mutation' is the only \
object type allowed in the mutations schema.

For mutation arguments, use 'input' types. For return values, use either \
entity types from the subgraph schema or interfaces.`
```

I actually think that we should allow other, non-entity types as return values here, and should allow users to define these types in the mutations schema.

The combined schema validation would have to become a little smarter to handle these. One trick could be to transform the type definitions in the mutations schema to add a `@mutations` directive to all of them. In the combined schema validation, we could then ignore all types with this directive.

Reply:
Re @mutations directive: This would require changes within the graph-node to ignore these new types correct?

Reply:
The error message has been updated with the version you wrote, thank you for simplifying the wording!